src/app/core/services/app-server/app-server.service.ts

```diff
@@ -0,0 +1,172 @@
+import { Injectable } from '@angular/core'
+import * as moment from 'moment-timezone'
+import * as Swagger from 'swagger-client'
+
+import {
+  DefaultAppServerURL,
+  DefaultRequestJSONContentType
+} from '../../../../assets/data/defaultConfig'
+import { ConfigKeys } from '../../../shared/enums/config'
+import { StorageKeys } from '../../../shared/enums/storage'
+import { RemoteConfigService } from '../config/remote-config.service'
+import { SubjectConfigService } from '../config/subject-config.service'
+import { LocalizationService } from '../misc/localization.service'
+import { LogService } from '../misc/log.service'
+import { StorageService } from '../storage/storage.service'
+import { TokenService } from '../token/token.service'
+
+@Injectable()
+export class AppServerService {
+  private API_DOCS_URL = '/v3/api-docs'
+
+  RADAR_USER_CONTROLLER = 'radar-user-controller'
+  RADAR_PROJECT_CONTROLLER = 'radar-project-controller'
+  MAX_API_RETRIES = 10
+  apiClient: any
+
+  constructor(
+    public storage: StorageService,
+    public subjectConfig: SubjectConfigService,
+    public logger: LogService,
+    public remoteConfig: RemoteConfigService,
+    public localization: LocalizationService,
+    private token: TokenService
+  ) {}
+
+  init() {
+    return (this.apiClient
+      ? Promise.resolve()
+      : this.initApiClient()
+    ).then(() => this.checkProjectAndSubjectExistElseCreate())
+  }
+
+  initApiClient() {
+    return Promise.all([this.getAppServerURL(), this.token.getTokens()]).then(
+      ([url, tokens]) =>
+        Swagger({
```

Comment:
I think we decided to use the swagger client because at the time the app server was in heavy development so in case the app server endpoints change (or in case of other api changes), this would allow us to make minimal changes to the questionnaire code, since the api docs will be loaded automatically. I think now we can switch to basic HTTP requests. What do you think @yatharthranjan?

Reply:
Yes, i guess that's a good point. I think as nivethika mentioned since the models are already there, it does not provide much value, i am happy with converting to the Angular client.